internal/matrix/database/matrix_queries.go

```diff
@@ -1,28 +1,76 @@
 package database
 
+import "fmt"
+
 const (
-	createMatrix    = "create matrix"
-	deleteMatrix    = "delete matrix by uuid"
-	getMatrix       = "get matrix by uuid"
-	getCourseMatrix = "get course and matrix by uuid"
-	listMatrix      = "list matrices"
-	updateMatrix    = "update matrix by uuid"
-	addSubject      = "adds subject to matrix"
-	removeSubject   = "remove subject from matrix"
+	returningMatrixColumns = `matrices.uuid, matrices.code, matrices.name, matrices.description,
+		matrices.created_at, matrices.updated_at`
+	// CREATE.
+	createMatrix = "create matrix"
+	addSubject   = "adds subject to matrix"
+	// READ.
+	getMatrix             = "get matrix by uuid"
+	getCourseMatrixExists = "get whether there is a relationship between course and matrix"
+	listMatrices          = "list matrices"
+	listCourseMatrices    = "list course matrices by course_uuid"
+	// UPDATE.
+	updateMatrix = "update matrix by uuid"
+	// DELETE.
+	deleteMatrix  = "delete matrix by uuid"
+	removeSubject = "remove subject from matrix"
 )
 
 func queriesMatrix() map[string]string {
 	return map[string]string{
-		createMatrix: "INSERT INTO matrices (name, description) VALUES ($1, $2) RETURNING *",
-		deleteMatrix: "UPDATE matrices SET deleted_at = NOW() WHERE uuid = $1",
-		getMatrix:    "SELECT * FROM matrices WHERE uuid = $1",
-		getCourseMatrix: `SELECT *
+		// CREATE.
+		createMatrix: fmt.Sprintf(`INSERT INTO matrices (
+				course_id, code, name, description
+			) SELECT courses.id, $2, $3, $4
+				FROM courses
+			WHERE courses.uuid = $1 AND courses.deleted_at IS NULL
+			RETURNING %s`, returningMatrixColumns),
+		addSubject: "INSERT INTO matrix_subjects (matrix_id, subject_id) VALUES($1, $2)",
+		// READ.
+		getMatrix: fmt.Sprintf(`SELECT
+				courses.uuid AS course_uuid,
+				%s
+			FROM matrices
+			LEFT JOIN courses ON matrices.course_id = courses.id
+			WHERE matrices.uuid = $1 AND matrices.deleted_at IS NULL
+				AND courses.deleted_at IS NULL`, returningMatrixColumns),
+		getCourseMatrixExists: `SELECT EXISTS (
+			SELECT 1
+			FROM matrices
+			LEFT JOIN courses ON matrices.course_id = courses.id
```

Comment:
I think it makes more sense to use `inner join` in this case, to make it clear that exists is true when we have a match in this relationship

Reply:
https://github.com/sumelms/microservice-course/compare/67c4ee511ce15350cab029efa7efcf7487294284..c375ffc82ccd084302563c096adf9ecf88621c4f